dffml/source/file.py

```diff
@@ -66,6 +79,13 @@ def __repr__(self):
             elif self.filename[::-1].startswith(('.xz')[::-1]) or \
                     self.filename[::-1].startswith(('.lzma')[::-1]):
                 close = lzma.open(self.filename, 'wt')
+            elif self.filename[::-1].startswith(('.zip')[::-1]):
+                @contextmanager
+                def closer_helper():
+                    with zipfile.ZipFile(self.filename, 'w') as archive:
+                        with io.TextIOWrapper(archive.open(str(self.filename[:-3]) + 'csv', 'w')) as fd:
```

Comment:
I have to get the type of file source but this is discarded after the corresponding source instance is created. i.e, in ```-sources``` we mention ```csv=training.zip``` but ```csv``` is discarded in ```parseSourcesAction``` as soon as an object of ```CSVSource``` is created. To avoid mentioning csv directly i've derived the type from class name for now. @pdxjohnny  is this okay?

Reply:
Yes that works. However, if some uses another source on the same `.zip` aka:

```console
-sources csv=test.zip
```

and then later

```console
-sources json=test.zip
```

The `open_helper` will iterate over the files in the zip archive and may end up loading the wrong file type on open.

Once `Source` and `Model` (#46, #47) have been updated to get passed config objects then we could add more parameters like which file within the zipfile to use. However, until then, I'd say replacing that saved filename with

```python
                        with io.TextIOWrapper(archive.open('internalfile', 'w')) as fd:
```

is what we should go with.

Reply:
Okay :+1: